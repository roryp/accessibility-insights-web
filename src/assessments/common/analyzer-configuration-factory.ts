// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Messages } from 'common/messages';
import { RuleAnalyzerConfiguration } from 'injected/analyzers/analyzer';
import { ScannerUtils } from 'injected/scanner-utils';

export class AnalyzerConfigurationFactory {
    public static forScanner(base: Partial<RuleAnalyzerConfiguration>): RuleAnalyzerConfiguration {
        const defaultValues: Partial<RuleAnalyzerConfiguration> = {
            resultProcessor: (scanner: ScannerUtils) => scanner.getAllCompletedInstances,
            telemetryProcessor: telemetryFactory => telemetryFactory.forAssessmentRequirementScan,
            analyzerMessageType: Messages.Assessment.AssessmentScanCompleted,
        };

        return {
            ...defaultValues,
            ...base,
        } as RuleAnalyzerConfiguration;
    }
}
