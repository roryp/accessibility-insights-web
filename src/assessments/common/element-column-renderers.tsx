// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { AssessmentInstanceDetailsColumn } from 'DetailsView/components/assessment-instance-details-column';
import * as React from 'react';

export function onRenderPathColumn(item: InstanceTableRow): JSX.Element {
    let textContent = '';
    if (item.instance.target) {
        textContent = item.instance.target.join(';');
    }

    return (
        <AssessmentInstanceDetailsColumn
            background={null}
            labelText={null}
            textContent={textContent}
            tooltipId={null}
            customClassName="not-applicable"
        />
    );
}

export function onRenderSnippetColumn(item: InstanceTableRow): JSX.Element {
    return (
        <AssessmentInstanceDetailsColumn
            background={null}
            labelText={null}
            textContent={item.instance.html}
            tooltipId={null}
            customClassName="not-applicable"
        />
    );
}
