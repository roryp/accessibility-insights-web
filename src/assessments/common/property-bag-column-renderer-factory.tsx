// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { AssessmentInstanceRowData } from 'DetailsView/components/assessment-instance-table';
import {
    propertyBagColumnRenderer,
    PropertyBagColumnRendererConfig,
} from './property-bag-column-renderer';

export class PropertyBagColumnRendererFactory {
    public static getRenderer<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: AssessmentInstanceRowData<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRenderer(item, configs);
        };
    }
}
