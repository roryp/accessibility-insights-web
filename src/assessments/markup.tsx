// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export function Tag(props: { tagName: string; isBold?: boolean }): JSX.Element {
    const isBold = props.hasOwnProperty('isBold') ? props.isBold : true;
    return isBold ? (
        <CodeTerm>&lt;{props.tagName}&gt;</CodeTerm>
    ) : (
        <Code>&lt;{props.tagName}&gt;</Code>
    );
}

export function Emphasis(props: { children: React.ReactNode }): JSX.Element {
    return <em>{props.children}</em>;
}

export function Term(props: { children: React.ReactNode }): JSX.Element {
    return <strong>{props.children}</strong>;
}

export function Code(props: { children: React.ReactNode }): JSX.Element {
    return <span className="insights-code">{props.children}</span>;
}

export function CodeBlock(props: { children: React.ReactNode }): JSX.Element {
    return <div className="insights-code">{props.children}</div>;
}

export const GreaterThanOrEqualTo = () => <span>&#8805;</span>;

export const NonBreakingSpace = () => <span>&nbsp;</span>;

export function CodeTerm(props: { children: React.ReactNode }): JSX.Element {
    return (
        <Term>
            <Code>{props.children}</Code>
        </Term>
    );
}
