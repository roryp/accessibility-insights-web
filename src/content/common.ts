// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { ContentCreator } from 'views/content/content-page';
import { GuidanceTitle as gt } from 'views/content/guidance-title';
import { link } from './link';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
