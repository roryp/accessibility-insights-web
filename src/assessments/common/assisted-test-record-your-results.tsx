// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export const AssistedTestRecordYourResults = () => (
    <li>
        Record your results:
        <ol>
            <li>
                Select <Markup.Term>Fail</Markup.Term> for any instances that do not meet the
                requirement.
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>. Or, after you have marked all
                failures, select <Markup.Term>Pass unmarked instances.</Markup.Term>
            </li>
        </ol>
    </li>
);
