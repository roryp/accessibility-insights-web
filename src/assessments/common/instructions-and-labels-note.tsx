// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

import * as Markup from '../markup';

export const InstructionsAndLabelsNotes = () => (
    <Markup.Emphasis>
        Note: For WCAG 2.1 compliance, both visible instructions and labels must be programmatically
        related to a widget. For WCAG 2.0 compliance, only visible instructions must be
        programmatically related to a widget.
    </Markup.Emphasis>
);
