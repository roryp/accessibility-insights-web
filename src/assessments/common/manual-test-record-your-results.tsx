// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export interface ManualTestRecordYourResultsProps {
    isMultipleFailurePossible: boolean;
}

export const ManualTestRecordYourResults = (props: ManualTestRecordYourResultsProps) => (
    <li>
        Record your results:
        <ol>
            <li>
                If you find {props.isMultipleFailurePossible ? 'any failures' : 'a failure'}, select{' '}
                <Markup.Term>Fail</Markup.Term>,
                {props.isMultipleFailurePossible
                    ? ' then add them as failure instances'
                    : ' then add the failure instance'}
                .
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>.
            </li>
        </ol>
    </li>
);
