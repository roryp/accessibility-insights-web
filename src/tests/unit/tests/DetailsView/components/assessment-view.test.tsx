// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';
import { Mock, Times } from 'typemoq';

import { AssessmentDefaultMessageGenerator } from 'assessments/assessment-default-message-generator';
import { AssessmentViewPropsBuilder } from 'tests/unit/tests/DetailsView/components/assessment-view-props-builder';
import { VisualizationType } from '../../../../../common/types/visualization-type';
import {
    AssessmentView,
    AssessmentViewProps,
} from '../../../../../DetailsView/components/assessment-view';
import { CreateTestAssessmentProvider } from '../../../common/test-assessment-provider';

describe('AssessmentViewTest', () => {
    const assessmentsProvider = CreateTestAssessmentProvider();
    const assessmentDefaultMessageGenerator = new AssessmentDefaultMessageGenerator();
    let builder: AssessmentViewPropsBuilder;

    beforeEach(() => {
        builder = new AssessmentViewPropsBuilder(
            assessmentsProvider,
            assessmentDefaultMessageGenerator,
        );
    });

    test('constructor', () => {
        const testObject = new AssessmentView({} as AssessmentViewProps);
        expect(testObject).toBeInstanceOf(React.Component);
    });

    test('render', () => {
        const props = builder.buildProps();

        const rendered = shallow(<AssessmentView {...props} />);
        expect(rendered.debug()).toMatchSnapshot();
    });

    test('componentDidMount', () => {
        const props = builder.buildProps();
        builder.updateHandlerMock.setup(u => u.onMount(props)).verifiable(Times.once());

        const testObject = new AssessmentView(props);

        testObject.componentDidMount();
        builder.verifyAll();
    });

    test('componentDidUpdate', () => {
        const prevProps = buildPrevProps();
        const props = builder.buildProps();
        const onAssessmentViewUpdateMock = Mock.ofInstance(
            (previousProps: AssessmentViewProps, currentProps: AssessmentViewProps) => {},
        );

        builder.updateHandlerMock.setup(u => u.update(prevProps, props)).verifiable(Times.once());
        builder.detailsViewExtensionPointMock
            .setup(d => d.apply(props.assessmentTestResult.definition.extensions))
            .returns(() => {
                return { onAssessmentViewUpdate: onAssessmentViewUpdateMock.object };
            })
            .verifiable(Times.once());
        onAssessmentViewUpdateMock.setup(o => o(prevProps, props)).verifiable(Times.once());

        const testObject = new AssessmentView(props);

        testObject.componentDidUpdate(prevProps);

        builder.verifyAll();
        onAssessmentViewUpdateMock.verifyAll();
    });

    test('componentWillUnmount', () => {
        const props = builder.buildProps();
        builder.updateHandlerMock.setup(u => u.onUnmount(props)).verifiable(Times.once());

        const testObject = new AssessmentView(props);

        testObject.componentWillUnmount();
        builder.verifyAll();
    });

    function buildPrevProps(): AssessmentViewProps {
        const prevStep = 'prevStep';
        const prevTest = -100 as VisualizationType;
        const prevProps = builder.buildProps();
        prevProps.assessmentNavState.selectedTestSubview = prevStep;
        prevProps.assessmentNavState.selectedTestType = prevTest;
        return prevProps;
    }
});
