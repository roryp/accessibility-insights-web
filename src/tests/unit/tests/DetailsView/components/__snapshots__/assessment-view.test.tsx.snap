// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentViewTest render 1`] = `
"<div className=\\"assessmentContent\\">
  <TargetChangeDialog deps={{...}} prevTab={{...}} newTab={{...}} />
  <div className=\\"assessmentTitle\\">
    <h1 className=\\"assessmentHeader\\">
      assessment 1
       
      <ContentLink deps={{...}} reference={[Function]} iconName=\\"info\\" />
    </h1>
  </div>
  <CollapsibleComponent header={{...}} content={{...}} containerClassName=\\"assessmentGettingStartedContainer\\" />
  <AssessmentViewMainContentExtensionPoint extensions={[undefined]} assessmentTestResult={{...}}>
    <div className=\\"assessmentRequirementsTitle\\">
      <h2 className=\\"assessmentRequirements\\">
        Requirements
      </h2>
    </div>
    <div className=\\"detailsViewAssessmentContent\\">
      <div className=\\"testStepsNavContainer\\">
        <TestStepsNav deps={{...}} ariaLabel=\\"Requirements\\" selectedTest={-1} selectedTestStep=\\"assessment-1-step-1\\" stepStatus={{...}} />
      </div>
      <div className=\\"testStepViewContainer\\">
        <TestStepView deps={{...}} isScanning={false} testStep={{...}} renderStaticContent={[undefined]} instancesMap={{...}} assessmentNavState={{...}} assessmentInstanceTableHandler={{...}} manualTestStepResultMap={{...}} assessmentsProvider={{...}} isStepEnabled={false} isStepScanned={false} assessmentDefaultMessageGenerator={{...}} featureFlagStoreData={{...}} pathSnippetStoreData={{...}} />
      </div>
    </div>
  </AssessmentViewMainContentExtensionPoint>
</div>"
`;
